R/graph.R

```diff
@@ -53,164 +130,176 @@ create_bigraph <- function(caobj,
       cgg_nn <- cgg_nn[,idx, drop = FALSE]
 
+
       if (isTRUE(prune_overlap)){
-        
-        overlap_mat <- calc_overlap( cc_adj = ccg_nn,
-                                     cg_adj = cgg_nn)
-        
+
+        cgg_nn <- calc_overlap( cc_adj = ccg_nn,
+                                     cg_adj = cgg_nn,
+                                     threshold = overlap)
+
         # For the case overlap = 1, all the genes are supposed to removed such that
         # the algorithm allows for clustering for cells without genes.
-        cgg_nn[overlap_mat <= overlap] <- 0
+        # cgg_nn[overlap_mat <= overlap] <- 0 # this step is done by cpp function to reduce the copy between cpp and R
         idx <- Matrix::colSums(cgg_nn) > 0
         cgg_nn <- cgg_nn[,idx, drop = FALSE]
 
       }
-    } 
-    
-    
+    }
+
+
 
     if(!is.null(marker_genes)){
-        
+
         cgg_nn <- cbind(cgg_nn, marker_knn)
-        
+
         # marker_dists <- marker_dists[,c(colnames(gene_dists), rownames(marker_dists))]
         # gene_dists <- cbind(rbind(gene_dists, marker_dists[,colnames(gene_dists)]), t(marker_dists))
         # gene_cell_assr <- rbind(gene_cell_assr, marker_assr)
-        
+
     }
-    
+
     # gene_idx <- which(rownames(caobj@prin_coords_rows) %in% colnames(cgg_nn))
-    
+
     gene_idx <- match(colnames(cgg_nn),
                       rownames(caobj@prin_coords_rows),
                       nomatch = NA_integer_)
     stopifnot(!any(is.na(gene_idx)))
-    
+
     ggg_nn = BiocNeighbors::findKNN(caobj@prin_coords_rows[gene_idx,],
                      k=k_g,
                      get.distance = FALSE,
                      BNPARAM=method,
-                     BPPARAM = BPPARAM)$index  
-    
+                     BPPARAM = BPPARAM)$index
+
     if (isTRUE(loops)){
         ggg_nn <- cbind(seq_len(nrow(ggg_nn)),
                         ggg_nn[, -ncol(ggg_nn), drop = FALSE])
     }
-    
-    ggg_nn <- indx_to_spmat(indx_mat = ggg_nn, 
-                            row_names = rownames(caobj@prin_coords_rows)[gene_idx], 
+
+    ggg_nn <- indx_to_spmat(indx_mat = ggg_nn,
+                            row_names = rownames(caobj@prin_coords_rows)[gene_idx],
                             col_names = rownames(caobj@prin_coords_rows)[gene_idx])
-    
-    
+
+
     # ggg_nn <- make_knn(gene_dists,
     #                    k = k_g,
     #                    decr = FALSE,
     #                    loops = loops)
-    
+
     if(isFALSE(calc_gene_cell_kNN)){
         gcg_nn <- Matrix::t(cgg_nn)
-        
+
+        if (k_gc != k_cg){
+          warning('The given values of k_gc and k_cg are different, But you set calc_cell_gene_kNN as FALSR,
+          so that the gene-cell graph adjacency matrix will be calculated as the transpose of cell-gene graph adjacency matrix.
+          This will ignore the k_gc value you have given. If you want to give k_gc and k_cg different values, set calc_cell_gene_kNN as TRUE please!')
```

Comment:
```
"The given values of k_gc and k_cg are different, but calc_cell_gene_kNN is FALSE. The the gene-cell graph adjacency matrix will be calculated as the transpose of cell-gene graph adjacency matrix. This will ignore the k_gc value you have given. If you want to give k_gc and k_cg different values, set calc_cell_gene_kNN as TRUE.
```